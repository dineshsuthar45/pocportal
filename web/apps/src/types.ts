import { InputProps, TypographyProps } from "@mui/material";
import { CSSProperties, ChangeEvent } from "react";

export interface IThemeSchema {
  palette: any;
  typography: any;
  components?: any;
}

export interface IGlobalTableProps {
  row: any;
  list: any;
}

export interface IGlobalButtonProps {
  label: any;
  sublabel?: string;
  disabled?: boolean;
  onClick?: () => void;
  style?: CSSProperties;
  type?: string;
}

export interface IGlobalImgProps {
  img: string;
  alt: string;
  style?: CSSProperties;
}

export interface IGlobaltextFieldProps {
  id?: string;
  value: string;
  label: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  name: string;
  errmsg?: string;
  type?: string;
  disabled?: boolean;
  style?: CSSProperties;
  placeholder?: string;
  showPassword?: boolean;
  InputProps?: InputProps;
}

export interface IGlobalTypoProps {
  id?: number;
  variant: TypographyProps["variant"];
  label: string;
  sublabel?: string;
  style?: CSSProperties;
}
