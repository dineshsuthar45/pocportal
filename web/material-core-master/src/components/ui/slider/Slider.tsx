import MuiSlider from "@mui/material/Slider";
import { useTheme } from "@mui/material/styles";
import { SliderProps } from "./Slider.types";

/**
 * Further properties can be found in MUI website at https://mui.com/material-ui/api/slider/
 */
export const MSlider = ({
  color = "secondary",
  orientation = "horizontal",
  size = "medium",
  disabled = false,
  track = "normal",
  ...props
}: SliderProps) => {
  const theme = useTheme();

  return (
    <MuiSlider
      className={`slider-root ${theme.palette.mode}`}
      color={color}
      orientation={orientation}
      size={size}
      disabled={disabled}
      track={track}
      {...props}
    />
  );
};

MSlider.defaultProps = {
  color: "secondary",
  orientation: "horizontal",
  size: "medium",
  disabled: false,
  track: "normal",
};
