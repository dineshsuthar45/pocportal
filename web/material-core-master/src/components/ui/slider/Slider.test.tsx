import { render } from "@testing-library/react";
import React from "react";
import { MSlider } from "./Slider";
import { SliderProps } from "./Slider.types";

describe("Slider component", () => {
  const renderComponent = () => render(<MSlider {...props} />);
  let props: SliderProps;

  beforeEach(() => {
    props = {};
  });

  it("should render correctly", () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
