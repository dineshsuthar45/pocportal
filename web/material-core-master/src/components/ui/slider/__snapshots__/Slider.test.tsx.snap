// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider component should render correctly 1`] = `
<div>
  <span
    class="MuiSlider-root MuiSlider-colorSecondary MuiSlider-sizeMedium slider-root light css-1diafny-MuiSlider-root"
  >
    <span
      class="MuiSlider-rail css-14pt78w-MuiSlider-rail"
    />
    <span
      class="MuiSlider-track css-1gv0vcd-MuiSlider-track"
      style="left: 0%; width: 0%;"
    />
    <span
      class="MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorSecondary MuiSlider-thumb MuiSlider-thumbSizeMedium MuiSlider-thumbColorSecondary css-frv3qg-MuiSlider-thumb"
      data-index="0"
      style="left: 0%;"
    >
      <input
        aria-orientation="horizontal"
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        data-index="0"
        max="100"
        min="0"
        step="1"
        style="border: 0px; height: 100%; margin: -1px; overflow: hidden; padding: 0px; position: absolute; white-space: nowrap; width: 100%; direction: ltr;"
        type="range"
        value="0"
      />
    </span>
  </span>
</div>
`;
